app/controllers/administrate/application_controller.rb

```diff
@@ -5,11 +5,13 @@ class ApplicationController < ActionController::Base
     def index
       authorize_resource(resource_class)
       search_term = params[:search].to_s.strip
-      resources = filter_resources(scoped_resource, search_term: search_term)
+      authorized_scope = authorize_scope(scoped_resource)
+      resources = filter_resources(authorized_scope, search_term: search_term)
       resources = apply_collection_includes(resources)
       resources = order.apply(resources)
       resources = paginate_resources(resources)
       page = Administrate::Page::Collection.new(dashboard, order: order)
+      page.context = self
```

Comment:
Following on my comment about `contextualize_resource`, I wonder if this is what it should be doing instead? Acting on the page instead of the resource here:

```suggestion
      contextualize_page(page)
```

Reply:
I think the Controller object is almost essential as a Context for the Page, so how about passing it to the initializer? Like the following:

```ruby
page = Administrate::Page::Collection.new(dashboard, context: self, order: order)
or
page = Administrate::Page::Collection.new(dashboard, controller: self, order: order)
```

Reply:
Initially that makes sense, but then the developers wouldn't be able to alter the context easily with a hook like `contextualize_page`. Having said that, perhaps there isn't a use case for that...? I have no idea yet. I'm experimenting now to see the possibilities of your PR.